Testcases may fail in an environment of insufficient performance

## Discussion

This issue is related to 
- #1659
- #1661 

My PC is using Ubuntu 20.04 LTS.
I use Docker to simulate the scene of insufficient performance and build project in the container
```bash
docker run \
--cpus=0.5 \
-m 4G \
--name maven-ejob -d \
-v /home/wuweijie/.m2/settings.xml:/usr/share/maven/conf/settings.xml \
-v$(pwd):/shardingsphere-elasticjob \
-w /shardingsphere-elasticjob \
maven:3.6.3-openjdk-8 mvn install
```

It failed to build module "elasticjob-lite-core".
```
Running org.apache.shardingsphere.elasticjob.lite.api.bootstrap.impl.OneOffJobBootstrapTest
Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 5.211 sec <<< FAILURE! - in org.apache.shardingsphere.elasticjob.lite.api.bootstrap.impl.OneOffJobBootstrapTest
assertExecute(org.apache.shardingsphere.elasticjob.lite.api.bootstrap.impl.OneOffJobBootstrapTest)  Time elapsed: 2.406 sec  <<< FAILURE!
java.lang.AssertionError: 

Expected: is <3>
     but: was <0>
	at org.apache.shardingsphere.elasticjob.lite.api.bootstrap.impl.OneOffJobBootstrapTest.assertExecute(OneOffJobBootstrapTest.java:80)
```

More logs:
```
[INFO] -----< org.apache.shardingsphere.elasticjob:elasticjob-lite-core >------
[INFO] Building elasticjob-lite-core 3.0.0-beta-SNAPSHOT                [23/39]
[INFO] --------------------------------[ jar ]---------------------------------
Downloading from alimaven: http://maven.aliyun.com/nexus/content/groups/public/org/quartz-scheduler/quartz/2.3.2/quartz-2.3.2.pom
Downloaded from alimaven: http://maven.aliyun.com/nexus/content/groups/public/org/quartz-scheduler/quartz/2.3.2/quartz-2.3.2.pom (9.0 kB at 18 kB/s)
Downloading from alimaven: http://maven.aliyun.com/nexus/content/groups/public/org/quartz-scheduler/quartz-parent/2.3.2/quartz-parent-2.3.2.pom
Downloaded from alimaven: http://maven.aliyun.com/nexus/content/groups/public/org/quartz-scheduler/quartz-parent/2.3.2/quartz-parent-2.3.2.pom (15 kB at 37 kB/s)
Downloading from alimaven: http://maven.aliyun.com/nexus/content/groups/public/com/mchange/mchange-commons-java/0.2.15/mchange-commons-java-0.2.15.pom
Downloaded from alimaven: http://maven.aliyun.com/nexus/content/groups/public/com/mchange/mchange-commons-java/0.2.15/mchange-commons-java-0.2.15.pom (3.7 kB at 11 kB/s)
Downloading from alimaven: http://maven.aliyun.com/nexus/content/groups/public/com/zaxxer/HikariCP-java7/2.4.13/HikariCP-java7-2.4.13.pom
Downloaded from alimaven: http://maven.aliyun.com/nexus/content/groups/public/com/zaxxer/HikariCP-java7/2.4.13/HikariCP-java7-2.4.13.pom (18 kB at 62 kB/s)
Downloading from alimaven: http://maven.aliyun.com/nexus/content/groups/public/org/quartz-scheduler/quartz/2.3.2/quartz-2.3.2.jar
Downloading from alimaven: http://maven.aliyun.com/nexus/content/groups/public/com/mchange/mchange-commons-java/0.2.15/mchange-commons-java-0.2.15.jar
Downloading from alimaven: http://maven.aliyun.com/nexus/content/groups/public/com/zaxxer/HikariCP-java7/2.4.13/HikariCP-java7-2.4.13.jar
Downloaded from alimaven: http://maven.aliyun.com/nexus/content/groups/public/com/zaxxer/HikariCP-java7/2.4.13/HikariCP-java7-2.4.13.jar (134 kB at 160 kB/s)
Downloaded from alimaven: http://maven.aliyun.com/nexus/content/groups/public/org/quartz-scheduler/quartz/2.3.2/quartz-2.3.2.jar (693 kB at 817 kB/s)
Downloaded from alimaven: http://maven.aliyun.com/nexus/content/groups/public/com/mchange/mchange-commons-java/0.2.15/mchange-commons-java-0.2.15.jar (624 kB at 489 kB/s)
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4:enforce (enforce-maven-version) @ elasticjob-lite-core ---
[INFO] 
[INFO] --- maven-enforcer-plugin:1.4:enforce (enforce-banned-dependencies) @ elasticjob-lite-core ---
[INFO] 
[INFO] --- maven-checkstyle-plugin:3.1.0:check (validate) @ elasticjob-lite-core ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.5:prepare-agent (default) @ elasticjob-lite-core ---
[INFO] argLine set to -javaagent:/root/.m2/repository/org/jacoco/org.jacoco.agent/0.8.5/org.jacoco.agent-0.8.5-runtime.jar=destfile=/shardingsphere-elasticjob/elasticjob-lite/elasticjob-lite-core/target/jacoco.exec
[INFO] 
[INFO] --- maven-remote-resources-plugin:1.5:process (process-resource-bundles) @ elasticjob-lite-core ---
[INFO] 
[INFO] --- maven-resources-plugin:2.7:resources (default-resources) @ elasticjob-lite-core ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /shardingsphere-elasticjob/elasticjob-lite/elasticjob-lite-core/src/main/resources
[INFO] Copying 3 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.3:compile (default-compile) @ elasticjob-lite-core ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-resources-plugin:2.7:testResources (default-testResources) @ elasticjob-lite-core ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 3 resources
[INFO] Copying 3 resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.3:testCompile (default-testCompile) @ elasticjob-lite-core ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:2.19.1:test (default-test) @ elasticjob-lite-core ---

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running org.apache.shardingsphere.elasticjob.lite.internal.instance.TriggerListenerManagerTest
Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 20.095 sec - in org.apache.shardingsphere.elasticjob.lite.internal.instance.TriggerListenerManagerTest
Running org.apache.shardingsphere.elasticjob.lite.internal.instance.ShutdownListenerManagerTest
Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.704 sec - in org.apache.shardingsphere.elasticjob.lite.internal.instance.ShutdownListenerManagerTest
Running org.apache.shardingsphere.elasticjob.lite.internal.instance.InstanceServiceTest
Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.798 sec - in org.apache.shardingsphere.elasticjob.lite.internal.instance.InstanceServiceTest
Running org.apache.shardingsphere.elasticjob.lite.internal.instance.InstanceNodeTest
Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.09 sec - in org.apache.shardingsphere.elasticjob.lite.internal.instance.InstanceNodeTest
Running org.apache.shardingsphere.elasticjob.lite.internal.server.ServerNodeTest
Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.003 sec - in org.apache.shardingsphere.elasticjob.lite.internal.server.ServerNodeTest
Running org.apache.shardingsphere.elasticjob.lite.internal.server.ServerServiceTest
Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.489 sec - in org.apache.shardingsphere.elasticjob.lite.internal.server.ServerServiceTest
Running org.apache.shardingsphere.elasticjob.lite.internal.guarantee.GuaranteeNodeTest
Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.007 sec - in org.apache.shardingsphere.elasticjob.lite.internal.guarantee.GuaranteeNodeTest
Running org.apache.shardingsphere.elasticjob.lite.internal.guarantee.GuaranteeServiceTest
Tests run: 14, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.592 sec - in org.apache.shardingsphere.elasticjob.lite.internal.guarantee.GuaranteeServiceTest
Running org.apache.shardingsphere.elasticjob.lite.internal.guarantee.GuaranteeListenerManagerTest
Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.914 sec - in org.apache.shardingsphere.elasticjob.lite.internal.guarantee.GuaranteeListenerManagerTest
Running org.apache.shardingsphere.elasticjob.lite.internal.election.LeaderServiceTest
Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.302 sec - in org.apache.shardingsphere.elasticjob.lite.internal.election.LeaderServiceTest
Running org.apache.shardingsphere.elasticjob.lite.internal.election.LeaderNodeTest
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.004 sec - in org.apache.shardingsphere.elasticjob.lite.internal.election.LeaderNodeTest
Running org.apache.shardingsphere.elasticjob.lite.internal.election.ElectionListenerManagerTest
Tests run: 11, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.603 sec - in org.apache.shardingsphere.elasticjob.lite.internal.election.ElectionListenerManagerTest
Running org.apache.shardingsphere.elasticjob.lite.internal.config.ConfigurationServiceTest
Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.997 sec - in org.apache.shardingsphere.elasticjob.lite.internal.config.ConfigurationServiceTest
Running org.apache.shardingsphere.elasticjob.lite.internal.config.RescheduleListenerManagerTest
Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.317 sec - in org.apache.shardingsphere.elasticjob.lite.internal.config.RescheduleListenerManagerTest
Running org.apache.shardingsphere.elasticjob.lite.internal.config.ConfigurationNodeTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.08 sec - in org.apache.shardingsphere.elasticjob.lite.internal.config.ConfigurationNodeTest
Running org.apache.shardingsphere.elasticjob.lite.internal.reconcile.ReconcileServiceTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.411 sec - in org.apache.shardingsphere.elasticjob.lite.internal.reconcile.ReconcileServiceTest
Running org.apache.shardingsphere.elasticjob.lite.internal.listener.ListenerManagerTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.623 sec - in org.apache.shardingsphere.elasticjob.lite.internal.listener.ListenerManagerTest
Running org.apache.shardingsphere.elasticjob.lite.internal.listener.RegistryCenterConnectionStateListenerTest
Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.968 sec - in org.apache.shardingsphere.elasticjob.lite.internal.listener.RegistryCenterConnectionStateListenerTest
Running org.apache.shardingsphere.elasticjob.lite.internal.listener.JobListenerTest
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.503 sec - in org.apache.shardingsphere.elasticjob.lite.internal.listener.JobListenerTest
Running org.apache.shardingsphere.elasticjob.lite.internal.failover.FailoverServiceTest
Tests run: 14, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.493 sec - in org.apache.shardingsphere.elasticjob.lite.internal.failover.FailoverServiceTest
Running org.apache.shardingsphere.elasticjob.lite.internal.failover.FailoverListenerManagerTest
Tests run: 11, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.906 sec - in org.apache.shardingsphere.elasticjob.lite.internal.failover.FailoverListenerManagerTest
Running org.apache.shardingsphere.elasticjob.lite.internal.failover.FailoverNodeTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.012 sec - in org.apache.shardingsphere.elasticjob.lite.internal.failover.FailoverNodeTest
Running org.apache.shardingsphere.elasticjob.lite.internal.schedule.SchedulerFacadeTest
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.205 sec - in org.apache.shardingsphere.elasticjob.lite.internal.schedule.SchedulerFacadeTest
Running org.apache.shardingsphere.elasticjob.lite.internal.schedule.JobRegistryTest
Tests run: 11, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.192 sec - in org.apache.shardingsphere.elasticjob.lite.internal.schedule.JobRegistryTest
Running org.apache.shardingsphere.elasticjob.lite.internal.schedule.LiteJobFacadeTest
Tests run: 17, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.41 sec - in org.apache.shardingsphere.elasticjob.lite.internal.schedule.LiteJobFacadeTest
Running org.apache.shardingsphere.elasticjob.lite.internal.schedule.JobTriggerListenerTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.188 sec - in org.apache.shardingsphere.elasticjob.lite.internal.schedule.JobTriggerListenerTest
Running org.apache.shardingsphere.elasticjob.lite.internal.schedule.JobScheduleControllerTest
Tests run: 25, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.397 sec - in org.apache.shardingsphere.elasticjob.lite.internal.schedule.JobScheduleControllerTest
Running org.apache.shardingsphere.elasticjob.lite.internal.snapshot.SnapshotServiceEnableTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 11.408 sec - in org.apache.shardingsphere.elasticjob.lite.internal.snapshot.SnapshotServiceEnableTest
Running org.apache.shardingsphere.elasticjob.lite.internal.snapshot.SnapshotServiceDisableTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 4.094 sec - in org.apache.shardingsphere.elasticjob.lite.internal.snapshot.SnapshotServiceDisableTest
Running org.apache.shardingsphere.elasticjob.lite.internal.setup.JobClassNameProviderFactoryTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.003 sec - in org.apache.shardingsphere.elasticjob.lite.internal.setup.JobClassNameProviderFactoryTest
Running org.apache.shardingsphere.elasticjob.lite.internal.setup.SetUpFacadeTest
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.496 sec - in org.apache.shardingsphere.elasticjob.lite.internal.setup.SetUpFacadeTest
Running org.apache.shardingsphere.elasticjob.lite.internal.setup.DefaultJobClassNameProviderTest
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.008 sec - in org.apache.shardingsphere.elasticjob.lite.internal.setup.DefaultJobClassNameProviderTest
Running org.apache.shardingsphere.elasticjob.lite.internal.storage.JobNodePathTest
Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.086 sec - in org.apache.shardingsphere.elasticjob.lite.internal.storage.JobNodePathTest
Running org.apache.shardingsphere.elasticjob.lite.internal.storage.JobNodeStorageTest
Tests run: 18, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 8.703 sec - in org.apache.shardingsphere.elasticjob.lite.internal.storage.JobNodeStorageTest
Running org.apache.shardingsphere.elasticjob.lite.internal.util.SensitiveInfoUtilsTest
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.01 sec - in org.apache.shardingsphere.elasticjob.lite.internal.util.SensitiveInfoUtilsTest
Running org.apache.shardingsphere.elasticjob.lite.internal.sharding.ExecutionServiceTest
Tests run: 17, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.187 sec - in org.apache.shardingsphere.elasticjob.lite.internal.sharding.ExecutionServiceTest
Running org.apache.shardingsphere.elasticjob.lite.internal.sharding.ShardingNodeTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec - in org.apache.shardingsphere.elasticjob.lite.internal.sharding.ShardingNodeTest
Running org.apache.shardingsphere.elasticjob.lite.internal.sharding.ShardingListenerManagerTest
Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.196 sec - in org.apache.shardingsphere.elasticjob.lite.internal.sharding.ShardingListenerManagerTest
Running org.apache.shardingsphere.elasticjob.lite.internal.sharding.ShardingServiceTest
Tests run: 16, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.695 sec - in org.apache.shardingsphere.elasticjob.lite.internal.sharding.ShardingServiceTest
Running org.apache.shardingsphere.elasticjob.lite.internal.sharding.ExecutionContextServiceTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.101 sec - in org.apache.shardingsphere.elasticjob.lite.internal.sharding.ExecutionContextServiceTest
Running org.apache.shardingsphere.elasticjob.lite.internal.sharding.MonitorExecutionListenerManagerTest
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.093 sec - in org.apache.shardingsphere.elasticjob.lite.internal.sharding.MonitorExecutionListenerManagerTest
Running org.apache.shardingsphere.elasticjob.lite.api.listener.DistributeOnceElasticJobListenerTest
Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.698 sec - in org.apache.shardingsphere.elasticjob.lite.api.listener.DistributeOnceElasticJobListenerTest
Running org.apache.shardingsphere.elasticjob.lite.api.bootstrap.impl.OneOffJobBootstrapTest
Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 5.211 sec <<< FAILURE! - in org.apache.shardingsphere.elasticjob.lite.api.bootstrap.impl.OneOffJobBootstrapTest
assertExecute(org.apache.shardingsphere.elasticjob.lite.api.bootstrap.impl.OneOffJobBootstrapTest)  Time elapsed: 2.406 sec  <<< FAILURE!
java.lang.AssertionError: 

Expected: is <3>
     but: was <0>
	at org.apache.shardingsphere.elasticjob.lite.api.bootstrap.impl.OneOffJobBootstrapTest.assertExecute(OneOffJobBootstrapTest.java:80)

Running org.apache.shardingsphere.elasticjob.lite.integrate.enable.ScheduleEnabledJobIntegrateTest
[ERROR] 2020-10-26 14:16:24,957 --test_one_off_job_execute_Worker-1-- [org.quartz.core.JobRunShell] Job DEFAULT.test_one_off_job_execute threw an unhandled Exception:  
org.apache.shardingsphere.elasticjob.reg.exception.RegException: java.lang.IllegalStateException: Client is not started
	at org.apache.shardingsphere.elasticjob.reg.exception.RegExceptionHandler.handleException(RegExceptionHandler.java:48)
	at org.apache.shardingsphere.elasticjob.reg.zookeeper.ZookeeperRegistryCenter.persist(ZookeeperRegistryCenter.java:219)
	at org.apache.shardingsphere.elasticjob.lite.internal.storage.JobNodeStorage.createJobNodeIfNeeded(JobNodeStorage.java:118)
	at org.apache.shardingsphere.elasticjob.lite.internal.sharding.ShardingService.resetShardingInfo(ShardingService.java:143)
	at org.apache.shardingsphere.elasticjob.lite.internal.sharding.ShardingService.shardingIfNecessary(ShardingService.java:120)
	at org.apache.shardingsphere.elasticjob.lite.internal.schedule.LiteJobFacade.getShardingContexts(LiteJobFacade.java:113)
	at org.apache.shardingsphere.elasticjob.executor.ElasticJobExecutor.execute(ElasticJobExecutor.java:91)
	at org.apache.shardingsphere.elasticjob.lite.internal.schedule.LiteJob.execute(LiteJob.java:35)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Caused by: java.lang.IllegalStateException: Client is not started
	at org.apache.curator.shaded.com.google.common.base.Preconditions.checkState(Preconditions.java:507)
	at org.apache.curator.CuratorZookeeperClient.getZooKeeper(CuratorZookeeperClient.java:139)
	at org.apache.curator.framework.imps.CuratorFrameworkImpl.getZooKeeper(CuratorFrameworkImpl.java:615)
	at org.apache.curator.framework.imps.CreateBuilderImpl$16.call(CreateBuilderImpl.java:1131)
	at org.apache.curator.framework.imps.CreateBuilderImpl$16.call(CreateBuilderImpl.java:1113)
	at org.apache.curator.RetryLoop.callWithRetry(RetryLoop.java:93)
	at org.apache.curator.framework.imps.CreateBuilderImpl.pathInForeground(CreateBuilderImpl.java:1110)
	at org.apache.curator.framework.imps.CreateBuilderImpl.protectedPathInForeground(CreateBuilderImpl.java:593)
	at org.apache.curator.framework.imps.CreateBuilderImpl.forPath(CreateBuilderImpl.java:583)
	at org.apache.curator.framework.imps.CreateBuilderImpl.forPath(CreateBuilderImpl.java:48)
	at org.apache.shardingsphere.elasticjob.reg.zookeeper.ZookeeperRegistryCenter.persist(ZookeeperRegistryCenter.java:212)
	... 8 common frames omitted
[ERROR] 2020-10-26 14:16:24,966 --test_one_off_job_execute_Worker-1-- [org.quartz.core.ErrorLogger] Job (DEFAULT.test_one_off_job_execute threw an exception. 
org.quartz.SchedulerException: Job threw an unhandled exception.
	at org.quartz.core.JobRunShell.run(JobRunShell.java:213)
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Caused by: org.apache.shardingsphere.elasticjob.reg.exception.RegException: java.lang.IllegalStateException: Client is not started
	at org.apache.shardingsphere.elasticjob.reg.exception.RegExceptionHandler.handleException(RegExceptionHandler.java:48)
	at org.apache.shardingsphere.elasticjob.reg.zookeeper.ZookeeperRegistryCenter.persist(ZookeeperRegistryCenter.java:219)
	at org.apache.shardingsphere.elasticjob.lite.internal.storage.JobNodeStorage.createJobNodeIfNeeded(JobNodeStorage.java:118)
	at org.apache.shardingsphere.elasticjob.lite.internal.sharding.ShardingService.resetShardingInfo(ShardingService.java:143)
	at org.apache.shardingsphere.elasticjob.lite.internal.sharding.ShardingService.shardingIfNecessary(ShardingService.java:120)
	at org.apache.shardingsphere.elasticjob.lite.internal.schedule.LiteJobFacade.getShardingContexts(LiteJobFacade.java:113)
	at org.apache.shardingsphere.elasticjob.executor.ElasticJobExecutor.execute(ElasticJobExecutor.java:91)
	at org.apache.shardingsphere.elasticjob.lite.internal.schedule.LiteJob.execute(LiteJob.java:35)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
	... 1 common frames omitted
Caused by: java.lang.IllegalStateException: Client is not started
	at org.apache.curator.shaded.com.google.common.base.Preconditions.checkState(Preconditions.java:507)
	at org.apache.curator.CuratorZookeeperClient.getZooKeeper(CuratorZookeeperClient.java:139)
	at org.apache.curator.framework.imps.CuratorFrameworkImpl.getZooKeeper(CuratorFrameworkImpl.java:615)
	at org.apache.curator.framework.imps.CreateBuilderImpl$16.call(CreateBuilderImpl.java:1131)
	at org.apache.curator.framework.imps.CreateBuilderImpl$16.call(CreateBuilderImpl.java:1113)
	at org.apache.curator.RetryLoop.callWithRetry(RetryLoop.java:93)
	at org.apache.curator.framework.imps.CreateBuilderImpl.pathInForeground(CreateBuilderImpl.java:1110)
	at org.apache.curator.framework.imps.CreateBuilderImpl.protectedPathInForeground(CreateBuilderImpl.java:593)
	at org.apache.curator.framework.imps.CreateBuilderImpl.forPath(CreateBuilderImpl.java:583)
	at org.apache.curator.framework.imps.CreateBuilderImpl.forPath(CreateBuilderImpl.java:48)
	at org.apache.shardingsphere.elasticjob.reg.zookeeper.ZookeeperRegistryCenter.persist(ZookeeperRegistryCenter.java:212)
	... 8 common frames omitted
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.693 sec - in org.apache.shardingsphere.elasticjob.lite.integrate.enable.ScheduleEnabledJobIntegrateTest
Running org.apache.shardingsphere.elasticjob.lite.integrate.enable.OneOffEnabledJobIntegrateTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.793 sec - in org.apache.shardingsphere.elasticjob.lite.integrate.enable.OneOffEnabledJobIntegrateTest
Running org.apache.shardingsphere.elasticjob.lite.integrate.disable.ScheduleDisabledJobIntegrateTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.293 sec - in org.apache.shardingsphere.elasticjob.lite.integrate.disable.ScheduleDisabledJobIntegrateTest
Running org.apache.shardingsphere.elasticjob.lite.integrate.disable.OneOffDisabledJobIntegrateTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.3 sec - in org.apache.shardingsphere.elasticjob.lite.integrate.disable.OneOffDisabledJobIntegrateTest

Results :

Failed tests: 
  OneOffJobBootstrapTest.assertExecute:80 
Expected: is <3>
     but: was <0>

Tests run: 306, Failures: 1, Errors: 0, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for elasticjob 3.0.0-beta-SNAPSHOT:
[INFO] 
[INFO] elasticjob ......................................... SUCCESS [03:51 min]
[INFO] elasticjob-api ..................................... SUCCESS [ 46.391 s]
[INFO] elasticjob-infra ................................... SUCCESS [  2.493 s]
[INFO] elasticjob-infra-common ............................ SUCCESS [ 49.204 s]
[INFO] elasticjob-registry-center ......................... SUCCESS [01:25 min]
[INFO] elasticjob-restful ................................. SUCCESS [ 50.498 s]
[INFO] elasticjob-ecosystem ............................... SUCCESS [  1.506 s]
[INFO] elasticjob-tracing ................................. SUCCESS [  1.795 s]
[INFO] elasticjob-tracing-api ............................. SUCCESS [ 40.611 s]
[INFO] elasticjob-error-handler ........................... SUCCESS [  0.688 s]
[INFO] elasticjob-error-handler-spi ....................... SUCCESS [  8.414 s]
[INFO] elasticjob-error-handler-type ...................... SUCCESS [  1.392 s]
[INFO] elasticjob-error-handler-general ................... SUCCESS [ 34.098 s]
[INFO] elasticjob-executor ................................ SUCCESS [  0.808 s]
[INFO] elasticjob-executor-kernel ......................... SUCCESS [ 36.905 s]
[INFO] elasticjob-executor-type ........................... SUCCESS [  1.297 s]
[INFO] elasticjob-simple-executor ......................... SUCCESS [ 37.091 s]
[INFO] elasticjob-dataflow-executor ....................... SUCCESS [ 44.505 s]
[INFO] elasticjob-script-executor ......................... SUCCESS [ 42.402 s]
[INFO] elasticjob-http-executor ........................... SUCCESS [ 37.905 s]
[INFO] elasticjob-tracing-rdb ............................. SUCCESS [ 40.979 s]
[INFO] elasticjob-lite .................................... SUCCESS [  1.420 s]
[INFO] elasticjob-lite-core ............................... FAILURE [01:43 min]
[INFO] elasticjob-lite-spring ............................. SKIPPED
[INFO] elasticjob-lite-spring-boot-starter ................ SKIPPED
[INFO] elasticjob-error-handler-email ..................... SKIPPED
[INFO] elasticjob-error-handler-wechat .................... SKIPPED
[INFO] elasticjob-error-handler-dingtalk .................. SKIPPED
[INFO] elasticjob-lite-spring-namespace ................... SKIPPED
[INFO] elasticjob-lite-lifecycle .......................... SKIPPED
[INFO] elasticjob-cloud ................................... SKIPPED
[INFO] elasticjob-cloud-common ............................ SKIPPED
[INFO] elasticjob-cloud-scheduler ......................... SKIPPED
[INFO] elasticjob-cloud-executor .......................... SKIPPED
[INFO] elasticjob-distribution ............................ SKIPPED
[INFO] elasticjob-src-distribution ........................ SKIPPED
[INFO] elasticjob-lite-distribution ....................... SKIPPED
[INFO] elasticjob-cloud-executor-distribution ............. SKIPPED
[INFO] elasticjob-cloud-scheduler-distribution ............ SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  15:20 min
[INFO] Finished at: 2020-10-26T14:16:36Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.19.1:test (default-test) on project elasticjob-lite-core: There are test failures.
[ERROR] 
[ERROR] Please refer to /shardingsphere-elasticjob/elasticjob-lite/elasticjob-lite-core/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :elasticjob-lite-core
```

--- a/elasticjob-lite/elasticjob-lite-core/src/test/java/org/apache/shardingsphere/elasticjob/lite/api/bootstrap/impl/OneOffJobBootstrapTest.java
+++ b/elasticjob-lite/elasticjob-lite-core/src/test/java/org/apache/shardingsphere/elasticjob/lite/api/bootstrap/impl/OneOffJobBootstrapTest.java
@@ -76,7 +76,7 @@ public final class OneOffJobBootstrapTest {
             counter.incrementAndGet();
         }, JobConfiguration.newBuilder("test_one_off_job_execute", SHARDING_TOTAL_COUNT).build());
         oneOffJobBootstrap.execute();
-        blockUtilFinish(oneOffJobBootstrap);
+        blockUtilFinish(oneOffJobBootstrap, counter);
         assertThat(counter.get(), is(SHARDING_TOTAL_COUNT));
         getJobScheduler(oneOffJobBootstrap).shutdown();
     }
@@ -105,9 +105,9 @@ public final class OneOffJobBootstrapTest {
     }
 
     @SneakyThrows
-    private void blockUtilFinish(final OneOffJobBootstrap oneOffJobBootstrap) {
+    private void blockUtilFinish(final OneOffJobBootstrap oneOffJobBootstrap, final AtomicInteger counter) {
         Scheduler scheduler = getScheduler(oneOffJobBootstrap);
-        while (!scheduler.isStarted() || !scheduler.getCurrentlyExecutingJobs().isEmpty()) {
+        while (0 == counter.get() || !scheduler.getCurrentlyExecutingJobs().isEmpty()) {
             Thread.sleep(100);
         }
     }

--- a/elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/JobSpringNamespaceWithTypeTest.java
+++ b/elasticjob-lite/elasticjob-lite-spring/elasticjob-lite-spring-namespace/src/test/java/org/apache/shardingsphere/elasticjob/lite/spring/namespace/job/JobSpringNamespaceWithTypeTest.java
@@ -25,7 +25,10 @@ import org.apache.shardingsphere.elasticjob.lite.spring.namespace.test.AbstractZ
 import org.apache.shardingsphere.elasticjob.reg.base.CoordinatorRegistryCenter;
 import org.junit.After;
 import org.junit.Test;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
 import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.util.ReflectionTestUtils;
 
 @ContextConfiguration(locations = "classpath:META-INF/job/withJobType.xml")
 public final class JobSpringNamespaceWithTypeTest extends AbstractZookeeperJUnit4SpringContextTests {
@@ -35,14 +38,22 @@ public final class JobSpringNamespaceWithTypeTest extends AbstractZookeeperJUnit
     @Resource
     private CoordinatorRegistryCenter regCenter;
     
+    private Scheduler scheduler;
+
     @After
-    public void tearDown() {
-        JobRegistry.getInstance().shutdown(scriptJobName);
+    public void tearDown() throws SchedulerException {
+        while (!scheduler.getCurrentlyExecutingJobs().isEmpty()) {
+            BlockUtils.waitingShortTime();
+        }
+        JobRegistry.getInstance().getJobScheduleController(scriptJobName).shutdown();
     }
     
     @Test
-    public void jobScriptWithJobTypeTest() {
-        BlockUtils.sleep(1000L);
-        assertTrue(regCenter.isExisted("/" + scriptJobName + "/sharding"));
+    public void jobScriptWithJobTypeTest() throws SchedulerException {
+        while (!regCenter.isExisted("/" + scriptJobName + "/sharding")) {
+            BlockUtils.waitingShortTime();
+        }
+        scheduler = (Scheduler) ReflectionTestUtils.getField(JobRegistry.getInstance().getJobScheduleController(scriptJobName), "scheduler");
+        assertTrue(scheduler.isStarted());
     }
 }

